docs/SECURITY_CONSIDERATIONS.md

```diff
@@ -19,6 +19,66 @@ assertion.repository_owner == 'my-github-org'
 Never use a "*" in an IAM Binding unless you absolutely know what you are doing!
 
 
+## Set up Repository & Branch Protection
```

Comment:
Branch Protection is deprecated and GitHub recommends rulesets now. That being said, I don't think the `auth` GitHub Action should be responsible for re-iterating GitHub security best practices.

I'm amenable to documenting "pull_request" vs "pull_request_target" things, but I don't want to become a documentation hub for GitHub repo security best practices. Perhaps we can link to GitHub documentation instead?